### DISCUSSION TITLE
New contract idea.

### DISCUSSION THREAD

**Discussion by panoptisDev** [mokshyaprotocol/aptos-nft-random-mint#11] [Ideas]
Hi i got an idea for a new contract. take a look at this frontend (https://github.com/panoptisDev/aptos-options-factory)
Turn this in to a guesing game. 
* put your bet of 1 APT (or any amout) in to a pool. Guess the price of $APT aptos at a certein date and time in the future, 
4 weeks, 3 weeks, 2 weeks, 7days or 4 days in the future. 
If someone guess the right price the contract pays out 90% of the pool to the one who guessing right price 4 weeks erlier,
and 75% of the pool to one alone winner who guessed right 3 weeks erlier. And so on.
if two or more players guess same exact price, the winning pot should be splitted between them. 
5%-10% of the pool should go to contract owner as service provider fee. the should be an option to create a pool of any Token 
deployed on aptos blockchain. 
best regards / panoptisDev
panoptis.network@gmail.com

**Comment by Subaashgautam** [comment]
this is an interesting idea

**Comment by newton-raphson** [comment]
Hy as aptos option factory has already done something similar, Can you mention the differentiating factor and include the math involved in making the bets and premium charges calculation associated with the bets ?
Regards,
Samundra Karki
